src/game/client/components/render_layer.cpp

```diff
@@ -1017,6 +1027,102 @@ void CRenderLayerQuads::Init()
 	RenderLoading();
 }
 
+void CRenderLayerQuads::CalculateClipping()
+{
+	// calculate clipping if not too expensive
+	if(!m_Grouped)
+		return;
+
+	// calculate envelope position offsets
+	ivec2 aEnvOffset[] = {{0, 0}, {0, 0}};
```

Comment:
The `vec` types should generally be used instead of separate horizontal/X/width and vertical/Y/height components (e.g. positions, sizes, offsets, speeds, literal vectors), not to group other non-vector space values together (min and max).

Accessing the `vec` components with the array operator like below is reserved for special occasions and defeats the purpose of using `vec` types to improve the readability.

Due to the added loop over the channels I think it doesn't really make sense to use `vec` anymore here.

```suggestion
	// Consider first two channels (X and Y offsets)
	int aEnvOffsetsMin[2] = {};
	int aEnvOffsetsMax[2] = {};
```